app/models/resource_group.rb

```diff
@@ -1,3 +1,8 @@
 class ResourceGroup < ApplicationRecord
-  has_many :vms
+  alias_attribute :images, :templates
+
+  has_many :vm_or_templates
+
+  has_many :vms,       -> { where(:template => false) }, :class_name => 'VmOrTemplate'
```

Comment:
Without changing the tests, if I do that I get: `uninitialized constant ResourceGroup::Template`